ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -1237,6 +1240,32 @@ public void getIndexSegmentFilesForLogSegmentTest() {
     }
   }
 
+  /**
+   * Tests {@link PersistentIndex#close()} can correctly cancel the scheduled persistor task and makes sure no persistor
+   * is running background after index closed.
+   * @throws IOException
+   * @throws StoreException
+   */
+  @Test
+  public void closeIndexToCancelPersistorTest() throws StoreException, InterruptedException {
+    state.index.close();
+    // re-initialize index by using mock scheduler (the intention is to speed up testing by using shorter period)
+    MockScheduledExecutorService mockScheduler =
+        new MockScheduledExecutorService(1, false, 0, 100, TimeUnit.MILLISECONDS);
+    mockScheduler.setContinueExistingPeriodicTasksAfterShutdownPolicy(false);
+    mockScheduler.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
+    state.initIndex(mockScheduler);
+    Thread.sleep(1);
+    // verify that the persistor task is successfully scheduled
+    assertEquals("The number of scheduled persistor task should be 1 after initialization", 1,
+        ((ThreadPoolExecutor) state.scheduler).getQueue().size());
+    state.index.close();
+    Thread.sleep(150);
```

Comment:
I think we should avoid looking at the internals of the `ThreadPoolExecutor`, since we don't understand it's usage of the queue completely, and so that we can wait deterministically instead of relying on sleeps. This would make the test less brittle.

For example, you can use `verify(mockScheduler , timeout(...), once()).scheduleAtFixedRate(...)` to replace the first sleep/queue check.

For the second  one, I would just check if the future was cancelled. You can use the `doAnswer` lambda to save a reference to the lambda, and you can check it on line 1270 by calling `Future.isCancelled()``